<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Launch the racecar model -->
  <include file="$(find f1tenth_simulator)/launch/racecar_model.launch"/>
  
  <!-- Default path for results if not passed -->
  <arg name="bfresults" default="/tmp/results.csv"/>
  
  <!-- Default path for configuration if not passed -->
  <arg name="bfconf" default="/tmp/default.yaml"/>

  <!-- RVIZ Display
      1 show normal drive mode
      2 show f1tenth simlator default perspective
      3 show both
      0 disable rviz display for bath simulation mode
  -->
  <arg name="show_rviz" default="0"/>
  <group if="$(eval arg('show_rviz') == 3)">
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find racer_world)/launch/rviz_config.rviz"/>
    <node pkg="rviz" type="rviz" name="rviz2" args="-d $(find f1tenth_simulator)/launch/simulator.rviz" output="screen"/>
  </group>
  <group if="$(eval arg('show_rviz') == 2)">
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find f1tenth_simulator)/launch/simulator.rviz" output="screen"/>
  </group>
  <group if="$(eval arg('show_rviz') == 1)">
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find racer_world)/launch/rviz_config.rviz"/>
  </group>

  <!-- Mode Override
      1 forces manual driving
      2 forces autonomous driving
      0 uses user input to enable the modes
  -->
  <arg name="mode_override" default="2"/>

  <!-- load robot description -->
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find racer_description)/urdf/racer.xacro'"/>

  <!-- load car controller, pass mode -->
  <include file="$(find car_control)/launch/car_control.launch">
      <arg name="mode_override" value="$(arg mode_override)"/>
  </include>

  <!--<include file="$(find vesc_driver)/launch/vesc_driver_node.launch"/>-->

  <!-- load -->
  <rosparam command="load" file="$(find vesc_sim)/config/car_config.yaml"/>

  <!--<include file="$(find vesc_ackermann)/launch/vesc_to_odom_node.launch"/>-->

<!--  <include file="$(find wallfollowing5)/launch/autonomous_driving_physical.launch">
    <arg name="topic_drive_param" value="/input/drive_param/autonomous" />
  </include>-->

  <node pkg="car_config" type="dynamic_physical_properties" name="dynamic_physical_properties" respawn="true" output="screen"/>
  
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen"/>

  <!--<include file="$(find hardware)/launch/static_wheel_publisher.launch"/>-->
</launch>
